src/main/java/com/dumbdogdiner/stickyapi/bukkit/command/StickyPluginCommand.java

```diff
@@ -0,0 +1,294 @@
+            }
+            onError(sender, commandLabel, arguments, code, variables);
+            return true;
+        } catch (Throwable ex) {
+            throw new CommandException("Unhandled exception executing command '" + commandLabel + "' in plugin "
+                    + owningPlugin.getDescription().getFullName(), ex);
+
+        }
+    }
+
+    public void onError(CommandSender sender, String commandLabel, Arguments arguments, ExitCode code, Map<String, String> vars) {
+        if (code == null)
+            return;
+        playSound(sender, code);
+        switch (code) {
+            case EXIT_ERROR_SILENT:
+            case EXIT_SUCCESS:
+            case EXIT_INFO:
+                return;
+            case EXIT_PERMISSION_DENIED:
+                sender.sendMessage(locale.translate("permission-denied", vars));
+                return;
+            case EXIT_BAD_SENDER:
+                sender.sendMessage(locale.translate("bad-sender", vars));
+                return;
+            case EXIT_MUST_BE_PLAYER:
+                sender.sendMessage(locale.translate("must-be-player", vars));
+                return;
+            case EXIT_COOLDOWN:
+                sender.sendMessage(locale.translate("cooldown", vars));
+                return;
+            case EXIT_INVALID_SYNTAX:
+                if (usageMessage.length() > 0) {
+                    for (String line : usageMessage.replace("<command>", commandLabel).split("\n")) {
+                        sender.sendMessage(line);
+                    }
+                }
+                return;
+            case EXIT_ERROR:
+            default:
+                sender.sendMessage(locale.translate("generic-error", vars));
+        }
+    }
+
+
+    /**
+     * Gets the owner of this PluginCommand
+     *
+     * @return Plugin that owns this command
+     */
+    @Override
+    @NotNull
+    public Plugin getPlugin() {
+        return getOwningPlugin();
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public abstract @NotNull List<String> tabComplete(@NotNull CommandSender sender, @NotNull String alias, @NotNull String[] args) throws IllegalArgumentException, CommandException;
+
+    /**
+     * <p>
+     * Stub method to perform validation for tabcomplete
+     * <p>
+     * If it is not present or returns null, will delegate to the current command
+     * executor if it implements {@link TabCompleter}. If a non-null list has not
+     * been found, will default to standard player name completion in
+     * {@link StickyPluginCommand#tabComplete(CommandSender, String, String[])}.
+     * <p>
+     *
+     * @throws IllegalArgumentException if sender, alias, or args is null
+     */
+
+    public void validateTabComplete(@NotNull CommandSender sender, @NotNull String alias,
+                                    @NotNull String[] args) throws IllegalArgumentException {
+        Validate.notNull(sender, "Sender cannot be null");
```

Comment:
Please use Preconditions from guava, rather than Validate. Guava is present in both Bukkit and Bungee and will work much better.